### PR TITLE
Task realization

### PR DESCRIPTION
**Pull request by victor110298**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,9 +7,48 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.2.0</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13-beta-2</version>
```
Comment (Anderwerty) [inline pom.xml]: image
https://mvnrepository.com/artifact/junit/junit
It is better to use `4.12` version.

File: pom.xml [hunk 2]
```diff
@@ -7,9 +7,48 @@
...
+            <scope>compile</scope>
+        </dependency>
     </dependencies>
-
-
 </project>
```
Comment (Anderwerty) [inline pom.xml]: Add a blank line at the end of the file.

File: src/main/java/store/Fruit.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package store;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.time.LocalDate;
+
+public class Fruit {
```
Comment (Anderwerty) [inline src/main/java/store/Fruit.java:14]: What do you think about using Lombok annotation?

File: src/main/java/store/FruitTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class FruitTest {
+    public class TradeShopTest {
+        private List<Fruit> firstListOfFruits;
```
Comment (Anderwerty) [inline src/main/java/store/FruitTest.java]: Order: constants, a blank line, fields

File: src/main/java/store/TypeOfFruit.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package store;
+
+public enum TypeOfFruit {
```
Comment (Anderwerty) [inline src/main/java/store/TypeOfFruit.java]: Naming is not right.

File: src/main/resources/log4j.properties.properties [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# ������� �����������
```
Comment (Anderwerty) [inline src/main/resources/log4j.properties.properties]: Remove comments in Russian.

File: src/main/resources/log4j.properties.properties [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+# ������� ��� ������ � �������
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+# ���� ��� ����� ����������� ��� ����
+log4j.appender.file.File=C:\Users\1\Desktop\homework_11
+# ��������� ������������ ������ ����� � ������
+log4j.appender.file.MaxFileSize=10MB
+# ������������� ������ ������ ����� � ����
+log4j.appender.file.layout=org.apache.log4j.PatternLayout
+log4j.appender.file.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
```
Comment (Anderwerty) [inline src/main/resources/log4j.properties.properties]: Add a blank line at the end of the file.

**Review by Anderwerty — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 8]
```diff
@@ -7,9 +7,48 @@
...
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.2</version>
+        </dependency>
+        <dependency>
```
Comment (Anderwerty) [inline pom.xml]: Do you use this dependency?

File: src/main/java/store/Fruit.java (hunk 3 above)
Comment (victor110298) [reply src/main/java/store/Fruit.java:14]: if it more good practise I can use it

Comment (Anderwerty) [reply src/main/java/store/Fruit.java:14]: It is just one of practice. Your teammates use Lombok and I think it'll good practice for your to use it.

File: pom.xml (hunk 8 above)
Comment (Anderwerty) [reply pom.xml]: Comment is not resolved. Remove this dependency if it is not used.

File: src/main/java/store/FruitType.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package store;
+
+public enum FruitType {
+    Cherry,
```
Comment (Anderwerty) [inline src/main/java/store/FruitType.java]: Naming is not right. 
 Apple -> APPLE
 Orange -> ORANGE
....

File: src/main/java/store/TradingStore.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        fruitList.clear();
+        addFruits(pathToJsonFile);
+    }
+
+    private boolean isSpoiled(Fruit fruit, LocalDate date) {
+        return fruit.getDate().plusDays(fruit.getShelfLife()).isAfter(date);
+    }
+
+    public List<Fruit> getSpoiledFruits(LocalDate date) {
+        List<Fruit> listOfSpoiled = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]:
```
return fruitList.stream()
  .filter(fruit -> isSpoiled(fruit, date))
 .collect(Collectors.toList());
```

File: src/main/java/store/TradingStore.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+    public List<Fruit> getAvailableFruits(LocalDate date) {
+        List<Fruit> listOfAvailableFruits = fruitList.stream()
+                .filter(fruit -> !isSpoiled(fruit, date))
+                .collect(Collectors.toList());
+        return listOfAvailableFruits;
+    }
+
+    public List<Fruit> getSpoiledFruits(LocalDate date, FruitType type) {
+        List<Fruit> listOfSpoiledFruits = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]: The same

File: src/main/java/store/TradingStore.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+                .filter(fruit -> fruit.getTypeOfFruit().equals(type))
+                .filter(fruit -> isSpoiled(fruit, date))
+                .collect(Collectors.toList());
+        return listOfSpoiledFruits;
+    }
+
+    public List<Fruit> getAvailableFruits(LocalDate date, FruitType type) {
+        List<Fruit> listOfAvailableFruits = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]: The same

File: src/main/java/store/TradingStore.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+                .filter(fruit -> fruit.getTypeOfFruit().equals(type))
+                .filter(fruit -> !isSpoiled(fruit, date))
+                .collect(Collectors.toList());
+        return listOfAvailableFruits;
+    }
+
+    public List<Fruit> getAddedFruits(LocalDate date) {
+        List<Fruit> listOfAvailableFruits = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]: The same

File: src/main/java/store/TradingStore.java [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+                .filter(fruit -> fruit.getDate().equals(date))
+                .collect(Collectors.toList());
+        return listOfAvailableFruits;
+    }
+
+    public List<Fruit> getAddedFruits(LocalDate date, FruitType typeOfFruit) {
+        List<Fruit> listOfAvailableFruits = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]: The same

File: src/main/java/store/TradingStore.java [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+    public List<Fruit> getSpoiledFruits(LocalDate date) {
+        List<Fruit> listOfSpoiled = fruitList.stream()
+                .filter(fruit -> isSpoiled(fruit, date))
+                .collect(Collectors.toList());
+        return listOfSpoiled;
+    }
+
+    public List<Fruit> getAvailableFruits(LocalDate date) {
+        List<Fruit> listOfAvailableFruits = fruitList.stream()
```
Comment (Anderwerty) [inline src/main/java/store/TradingStore.java]: The same

File: pom.xml (hunk 2 above)
Comment (Anderwerty) [reply pom.xml]: ?

**Review by Anderwerty — CHANGES_REQUESTED** [review]

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**